### PR TITLE
Property-based tests for the editions contract

### PR DESCRIPTION
**Pull request by dcastro**
This PR adds:
* Lorentz bindings for the editions contract
* Property-based tests for the invariants in the [editions contract spec](https://github.com/tqtezos/minter-sdk/blob/2ad6b155e27f9f5e5a0730f125875c821a36a1f0/packages/minter-contracts/ligo/src/minter_collection/editions/README.md).

Also took the opportunity to bump all the morley-related dependencies.

### CODE REVIEW

**Review by michaeljklein — APPROVED** [review]
LGTM

File: packages/minter-contracts/test-hs/Test/MinterCollection/Editions.hs [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+      pure $ EditionRunData 0 mempty numberOfEditions toDistribute
+
+    clevelandProp $ do
+      (initialStorage, admin) <- mkStorage $ adjustMaxEditionsPerRun [editionRun]
+      editions <- originateEditionsContract initialStorage
+      receivers <- createReceivers editionRun
+
+      withSender admin $ do
+        mintEditionRuns editions [editionRun]
+        distributeEditions editions [editionRun] [receivers]
```
[the remark this answers is not in the document]
Comment (dcastro) [reply packages/minter-contracts/test-hs/Test/MinterCollection/Editions.hs]: Done

File: packages/minter-contracts/src-hs/Lorentz/Contracts/FA2.hs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import Lorentz
+
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2I
+
+type OperatorStorage = BigMap OperatorKey ()
+
+data OperatorKey = OperatorKey
+  { osOwner :: Address
+  , osOperator :: Address
+  , osTokenId :: FA2I.TokenId
```
[the remark this answers is not in the document]
Comment (dcastro) [reply packages/minter-contracts/src-hs/Lorentz/Contracts/FA2.hs]: Ah nice, hadn't noticed that!

I happened to have a duplicate field name (both `EditionMetadata` and `EditionRunData` had a field named `editionInfo`), and start needing to add type annotations a lot, so I just renamed the latter to `dataEditionInfo`.

I could have also used [the `OverloadedLabels+DuplicateRecordFields` trick](https://www.reddit.com/r/haskell/comments/f7427x/overloadedlabels/), but renaming the field seemed like a simpler solution. If the problem gets worse in the future, we could consider that as a solution? Maybe even adding the `IsLabel` instance to `morley-prelude` :thinking:

File: packages/minter-contracts/src-hs/Util.hs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Util
+  (
+
+  ) where
```
Comment (pasqu4le) [inline packages/minter-contracts/src-hs/Util.hs:6]:
```suggestion
module Util () where
```

**Review by pasqu4le — APPROVED** [review]
LGTM too :ok_hand:

File: packages/minter-contracts/src-hs/Lorentz/Contracts/FA2.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Martoon-00) [reply packages/minter-contracts/src-hs/Lorentz/Contracts/FA2.hs]: Yeah, sounds good. A bit sad that `DuplicateRecordFields` causes such issues, do you understand why is it necessary?.

Using `generic-lens` seems quite good as a workaround if you fine the overall approach with prefix-less fields fine.

**Review by Martoon-00 — APPROVED** [review]

File: packages/minter-contracts/src-hs/Lorentz/Contracts/FA2.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dcastro) [reply packages/minter-contracts/src-hs/Lorentz/Contracts/FA2.hs]: > do you understand why is it necessary?.

I'm not sure what the technical reason is, but it's a [documented limitation](https://gitlab.haskell.org/ghc/ghc/-/wikis/records/overloaded-record-fields/duplicate-record-fields#selector-functions) of this feature :/

image

**Pull request merged**